### ISSUE TITLE
`nix bundle` _et al._ on derivation files directly

### ISSUE THREAD

**Issue by nrdxp** [NixOS/nix#7481]
**Is your feature request related to a problem? Please describe.**
In divnix/std-action, we are making a concerted effort to cleanly separate disparate concerns in a CI environment. There is the discovery phase (evaluation), the buid phase, the run phase, and the caching phase. In order to ensure that we don't violate boundaries between phases, we are going to great lengths to ensure that any and all Nix evaluation is done in the first phase (discovery).

This is more important than anything else since what is produced in this phase is parse in the build step, and what is ultimately cached depends on that, so it all traces back to the intial evaluation. If that is off, then we may miss caching our dependencies or spend extra time in a run phase doing Nix evaluation when we should be doing work.

We have a few commands that simply cannot be invoked on a derivation. Most of these have equivalents from the legacy `nix-*` but here are a few that don't:
```
nix bundle
nix profile # everything but install for some reason
# will add more if/when found
# ...
```

**Describe the solution you'd like**
Basically any command that makes sense to act on derivations directly should be able to do so to eliminate the need for evaluation if not necessary.

**Describe alternatives you've considered**
The eval-cache could be expanded to work in more situations to cover us where needed #6228 #4279

That said, tracking derivations themselves leaves us with no guess work. We don't have to ask ourselves, "our we missing the eval cache somehow."? We can just be rest assured we already have to result we want and act on it.

**Additional context**
https://github.com/divnix/std-action/issues/23

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
